aspnetcore/mvc/models/file-uploads/samples/5x/largeFiles/LargeFilesSample/Controllers/FileUploadController.cs

```diff
@@ -0,0 +1,108 @@
+        /// </summary>
+        /// <remarks>
+        /// Request to this action will not trigger any model binding or model validation,
+        /// because this is a no-argument action
+        /// </remarks>
+        /// <returns></returns>
+        [HttpPost]
+        [Route(nameof(UploadLargeFile))]
+        public async Task<IActionResult> UploadLargeFile()
+        {
+            var result = new JsonResponse();
+
+            var request = HttpContext.Request;
+
+            // validation of Content-Type
+            // 1. first, it must be a form-data request
+            // 2. a boundary should be found in the Content-Type
+            if (!request.HasFormContentType ||
+                !MediaTypeHeaderValue.TryParse(request.ContentType, out var mediaTypeHeader) ||
+                string.IsNullOrEmpty(mediaTypeHeader.Boundary.Value))
+            {
+                result.Message = "This Content-Type cannot be processed";
+                return BadRequest(result);
+            }
+
+            var reader = new MultipartReader(mediaTypeHeader.Boundary.Value, request.Body);
+            var section = await reader.ReadNextSectionAsync();
+
+            // This sample try to get the first file from request and save it
+            // Make changes according to your needs in actual use
+            while (section != null)
+            {
+                var hasContentDispositionHeader = ContentDispositionHeaderValue.TryParse(section.ContentDisposition,
+                    out var contentDisposition);
+
+                if (hasContentDispositionHeader && contentDisposition.DispositionType.Equals("form-data") &&
+                    !string.IsNullOrEmpty(contentDisposition.FileName.Value))
+                {
+                    // Don't trust any file name, file extension, and file data from the request unless you trust them completely
+                    // Otherwise, it is very likely to cause problems such as virus uploading, disk filling, etc
+                    // In short, it is necessary to restrict and verify the upload
+
+                    // This example only introduces how to save the uploaded data,but ignores the situations mentioned above.
+                    // Here, the file name is considered trusty by default.
+
+                    // Get the file extension, and combine a random file name with it
+                    var fileName = Guid.NewGuid().ToString("N") + Path.GetExtension(contentDisposition.FileName.Value);
+                    var saveToPath = Path.Combine(AppContext.BaseDirectory, "files", fileName);
+
+                    using (var targetStream = System.IO.File.Create(saveToPath))
+                    {
+                        await section.Body.CopyToAsync(targetStream);
+                    }
+
+                    return Ok(new
+                    {
+                        message = "Uploaded successfully!",
+                        fileUrl = $"/files/{fileName}"
+                    });
```

Comment:
```suggestion
                    return Ok();
```

I don't think we want to demonstrate making the file downloadable in this sample. If we had to, we would use CreatedAtResult